wled00/FX.cpp

```diff
@@ -3116,6 +3116,398 @@ static uint16_t rolling_balls(void) {
+      SEGMENT.setPixelColor(character[PACMAN].pos, character[PACMAN].color);    // draw PacMan
+      SEGMENT.setPixelColor(character[PACMAN].pos-1, BLACK);                    //   and the black dot behind him
+      character[PACMAN].pos += 1;                                               // update PacMan's position forwards for the next frame draw
+
+      if (character[PACMAN].topPos < character[PACMAN].pos)                     // keep track of the top (farthest) position of the PacMan character
+        character[PACMAN].topPos = character[PACMAN].pos;
+
+      for (int i = 1; i < numGhosts + 1; i++) {                                 // ...draw the 4 ghosts (and the black dot behind each ghost)
+        SEGMENT.setPixelColor(character[i].pos, character[i].color);
+        SEGMENT.setPixelColor(character[i].pos-1, BLACK);
+        character[i].pos += 1;                                                  // update their positions forwards for the next frame draw
+      }
+    }
+  }
+  else {                                                                        // Going backward (after PacMan ate a power dot)...
+    if (SEGENV.aux1 % map(SEGMENT.speed, 0, 255, 20, 1) == 0) {                 // User-selectable speed of PacMan and the Ghosts
+      SEGMENT.setPixelColor(character[PACMAN].pos+1, BLACK);                    // TODO: This causes a white dot to be deleted right after the powerdot when it's eaten; not sure how to fix it.
+      SEGMENT.setPixelColor(character[PACMAN].pos, character[PACMAN].color);    // draw PacMan
+      SEGMENT.setPixelColor(character[PACMAN].pos-1, BLACK);                    //   and the black dot behind him
+      character[PACMAN].pos -= 1;                                               // update PacMan's position backwards for the next frame draw
+
+      for (int i = 1; i < numGhosts + 1; i++) {                                 // ...draw the 4 ghosts (and black dots surrounding each ghost)
+
+        // if the ghost is blue and nearing the beginning of the strip, blink the ghosts.  TODO: The ghosts blink too fast or too slow, depending on the Speed slider's value; not sure how to fix it. 
+        if (character[1].blue && (character[PACMAN].pos <= startBlinkingGhostsLED)) {
+          if (character[i].color == BLUE)
+              character[i].color = BLACK;
+          else
+              character[i].color = BLUE;
+        }
+
+        SEGMENT.setPixelColor(character[i].pos+1, BLACK);
+        SEGMENT.setPixelColor(character[i].pos, character[i].color);
+        SEGMENT.setPixelColor(character[i].pos-1, BLACK);
+        character[i].pos -= 1;                                                  // update their positions backwards for the next frame draw
+      }
+    }
+  }
+
+  return FRAMETIME;
+}
+static const char _data_FX_MODE_PACMAN[] PROGMEM = "PacMan@Speed,# of Power Dots,,,,White dots,Compact dots,;,,;!;1;m12=0,o1=1";
```

Comment:
There is no need to specify `;,,;` if there are no color selectors. jus use `;;`.